### PR TITLE
Oppfolging periode

### PR DESCRIPTION
**Pull request by dragutin-nav**

### CODE REVIEW

**Comment by dragutin-nav** [comment]
@stianStensli its rebased on dev now

File: src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingPeriodeService.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+@Service
+@RequiredArgsConstructor
+public class OppfolgingPeriodeService extends KafkaCommonConsumerService<SisteOppfolgingsperiodeV1> {
+    private final OppfolgingStartetService oppfolgingStartetService;
+    private final OppfolgingAvsluttetService oppfolgingAvsluttetService;
+
+    @Override
+    protected void behandleKafkaMeldingLogikk(SisteOppfolgingsperiodeV1 sisteOppfolgingsperiod) {
+        if (sisteOppfolgingsperiod == null || sisteOppfolgingsperiod.getAktorId().isEmpty() || sisteOppfolgingsperiod.getStartDato() == null) {
+            log.warn("Ugyldig data for siste oppfolging periode");
```
Comment (stianStensli) [inline src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingPeriodeService.java:20]: kan we logg getAktorId? 
feks. "Ugyldig data for siste oppfolgingperiode, på bruker: {}"

File: src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingStartetService.java [hunk 2]
```diff
@@ -29,11 +30,14 @@
     @Override
     public void behandleKafkaMeldingLogikk(OppfolgingStartetDTO dto) {
         mapAktoerTilPersonId(dto.getAktorId());
+        startOppfolging(dto.getAktorId(), dto.getOppfolgingStartet());
+    }
```
Comment (stianStensli) [inline src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingStartetService.java:34]: is this method still in use? behandleKafkaMeldingLogikk

Comment (stianStensli) [reply src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingStartetService.java:34]: I guess that if this one can be removed, then we can also remove the one in oppfølging avsluttet

File: src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingPeriodeService.java (hunk 1 above)
Comment (dragutin-nav) [reply src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingPeriodeService.java:20]: Yep, just a sec to fix that

File: src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingStartetService.java (hunk 2 above)
Comment (dragutin-nav) [reply src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingStartetService.java:34]: Yep, agree...will be fixed

File: src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingPeriodeService.java [hunk 3]
```diff
@@ -17,7 +17,12 @@
     @Override
     protected void behandleKafkaMeldingLogikk(SisteOppfolgingsperiodeV1 sisteOppfolgingsperiod) {
         if (sisteOppfolgingsperiod == null || sisteOppfolgingsperiod.getAktorId().isEmpty() || sisteOppfolgingsperiod.getStartDato() == null) {
-            log.warn("Ugyldig data for siste oppfolging periode");
+            log.warn("Ugyldig data for siste oppfolging periode på bruker: " + sisteOppfolgingsperiod.getAktorId());
+            return;
+        }
+
+        if (sisteOppfolgingsperiod.getStartDato() != null && sisteOppfolgingsperiod.getSluttDato() != null && sisteOppfolgingsperiod.getStartDato().isAfter(sisteOppfolgingsperiod.getSluttDato())) {
+            log.warn("Ugyldig start/slutt dato for siste oppfolging periode på bruker: " + sisteOppfolgingsperiod.getAktorId());
```
Comment (stianStensli) [inline src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingPeriodeService.java:25]: This is very bad, maybe log error?

File: src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingStartetService.java [hunk 4]
```diff
@@ -3,61 +3,26 @@
...
-    }
 
     public void startOppfolging(AktorId aktorId, ZonedDateTime oppfolgingStartetDate) {
         oppfolgingRepository.settUnderOppfolging(aktorId, oppfolgingStartetDate);
         oppfolgingRepositoryV2.settUnderOppfolging(aktorId, oppfolgingStartetDate);
         opensearchIndexer.indekser(aktorId);
         log.info("Bruker {} har startet oppfølging: {}", aktorId, oppfolgingStartetDate);
     }
-
-    private void mapAktoerTilPersonId(AktorId aktorId) {
```
Comment (stianStensli) [inline src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingStartetService.java:39]: This code should not be removed. this is the code that makes data from arena mappable to aktoerId

Comment (stianStensli) [reply src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingStartetService.java:39]: Maybe move it to the new service?

Comment (stianStensli) [reply src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingStartetService.java:39]: Or to startOppfolging method

File: src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingPeriodeService.java (hunk 3 above)
Comment (dragutin-nav) [reply src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingPeriodeService.java:25]: true true

File: src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingStartetService.java (hunk 4 above)
Comment (dragutin-nav) [reply src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingStartetService.java:39]: so this mapping should be on oppfolging start..

Comment (stianStensli) [reply src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingStartetService.java:39]: Yes, but it does not matter if we also map on oppfølging avsluttet. 
The plan is to do this mapping based on a topic from PDL, but that work is a bit far away

File: src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingStartetService.java [hunk 5]
```diff
@@ -19,21 +19,19 @@
...
-    @Override
-    public void behandleKafkaMeldingLogikk(OppfolgingStartetDTO dto) {
-        mapAktoerTilPersonId(dto.getAktorId());
-
-        oppfolgingRepository.settUnderOppfolging(dto.getAktorId(), dto.getOppfolgingStartet());
-        oppfolgingRepositoryV2.settUnderOppfolging(dto.getAktorId(), dto.getOppfolgingStartet());
-        opensearchIndexer.indekser(dto.getAktorId());
-        log.info("Bruker {} har startet oppfølging: {}", dto.getAktorId(), dto.getOppfolgingStartet());
+    public void startOppfolging(AktorId aktorId, ZonedDateTime oppfolgingStartetDate) {
+        mapAktoerTilPersonId(aktorId);
```
Comment (dragutin-nav) [inline src/main/java/no/nav/pto/veilarbportefolje/oppfolging/OppfolgingStartetService.java:30]: @stianStensli is it ok like this?

**Review by stianStensli — APPROVED** [review]
Ready for test 🎉

**Pull request merged**